modules/routes/calendar/calendar_event_route.py

```diff
@@ -0,0 +1,77 @@
+from flask import request, jsonify
+from app import app
+from modules.models.account import AccountType
+from modules.routes.calendar.calendar_event_schema import (
+    CalendarEventSchema,
+    DateRangeSchema,
+
+)
+from modules.utils.db import db
+from modules.utils.route_utils import authenticate, fetch_session, protected_route
+
+@app.route("/calendar/<str:course_id>", methods=["POST"])
+@protected_route(roles=[AccountType.Admin, AccountType.Lecturer])
+def create_calendar_event(course_id: str):
+    body = CalendarEventSchema().load(request.get_json(force=True))
+
+    db_cursor = db.cursor()
+    try:
+        db_cursor.execute(
+            "INSERT INTO CalendarEvent (course_id, date, event_name, event_no) VALUES (%s, %s, %s, %s)",
+            (body["course_id"], body["date"], body["event_name"]),
+        )
+        db.commit()
+        return jsonify({"message": "Calendar event created successfully"}), 201
+    except Exception as e:
+        db.rollback()
+        return jsonify({"message": f"Failed to create calendar event: {str(e)}"}), 500
+
+@app.route("/calendar/events/course/<string:course_id>", methods=["GET"])
+def get_course_calendar_events(course_id):
```

Comment:
This should be a protected route. I'm almost certain I told you all the endpoints in this route should be protected...

```suggestion
@protected_route()
def get_course_calendar_events(course_id):
```

There should also be checks to see if a student is enrolled in the course or if a lecturer teaches the course.